compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -497,6 +497,10 @@ public void visit(BLangFunction funcNode, AnalyzerData data) {
         // TODO: Shouldn't this be done in symbol enter?
         //set function param flag to final
         funcNode.symbol.params.forEach(param -> param.flags |= Flags.FUNCTION_FINAL);
+        
+        if (funcNode.symbol.restParam != null) {
+            funcNode.symbol.restParam.flags |= Flags.FUNCTION_FINAL;
+        }
```

Comment:
This is not related to this change. What is the reason to introduce a new flag for this? Within the parameter symbol, there is already a flag to check whether it is a parameter. Can't we utilize that existing flag instead of introducing a new one? @MaryamZi

Reply:
Not sure which came first, but yeah, we may not need a separate flag. I wouldn't change this now though.